packages/x-data-grid/src/models/gridRowSelectionModel.ts

```diff
@@ -5,6 +5,61 @@ export type GridRowSelectionPropagation = {
   parents?: boolean;
 };
 
-export type GridInputRowSelectionModel = readonly GridRowId[] | GridRowId;
+export type GridRowSelectionModel = { type: 'include' | 'exclude'; ids: Set<GridRowId> };
 
-export type GridRowSelectionModel = readonly GridRowId[];
+interface SelectionManager {
+  data: Set<GridRowId>;
+  has(id: GridRowId): boolean;
+  select(id: GridRowId): void;
+  unselect(id: GridRowId): void;
+  size(): number;
+}
+class IncludeManager implements SelectionManager {
+  data: SelectionManager['data'];
+
+  constructor(model: GridRowSelectionModel) {
+    this.data = model.ids;
+  }
+
+  has: SelectionManager['has'] = (id) => this.data.has(id);
```

Comment:
Are the inline type annotations useful here? Can't TS infer that based on `implements`? It would also avoid allocating one new function for each method of each instance (which the arrow syntax class method does, unlike regular class methods).

Reply:
I left this for later, but with `implements` I had to redeclare each method's arguments: [playground](https://www.typescriptlang.org/play/?#code/C4TwDgpgBA4gTgSwCYCUD2B3AkkqBeKAZ2EQDsBzAbgChqFTgI4AzAQwGNoBlCAGwnbAEaUgFlWpVuSZQA3tShQAFq0IAKZAC5YiVJhwBKbQCM0afhJoBfWu16rCULKTsBXJBHGTpcKAgC2YPz+EAyOPPyCwmISUjLyiooq6sgGcgqJmXAQwK5wpFAAhIXINJk2NtRAA)
Unless there's another way to avoid that, I'll just go ahead and redeclare argument types.